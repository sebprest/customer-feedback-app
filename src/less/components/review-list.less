.review {
    margin-bottom: @margin-lg;
}
