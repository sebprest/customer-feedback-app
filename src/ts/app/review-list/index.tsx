import React from "react";
import { connect } from "react-redux";

import { ReviewsState } from "^/common/types";
import { StoreState } from "^/store/types";
import Review from "./review";

interface StateProps {
  reviews: ReviewsState;
}

type Props = StateProps;

export const ReviewList: React.FunctionComponent<Props> = React.memo(
  (props) => (
    <div className="review-list">
      {props.reviews.map((review, index) => (
        <Review review={review} key={index} />
      ))}
    </div>
  )
);

const mapStateToProps = ({ reviews }: StoreState): StateProps => {
  return {
    reviews,
  };
};

export default connect(mapStateToProps)(ReviewList);
