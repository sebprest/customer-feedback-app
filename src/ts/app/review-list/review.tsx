import React from "react";

import { Review as TReview } from "^/common/types";

type Props = {
  review: TReview;
};

export const Review: React.FunctionComponent<Props> = React.memo((props) => (
  <div className="review">
    {props.review.name} ({props.review.email}) - Rating: {props.review.rating}:
    <div className="review-comment">{props.review.comment}</div>
  </div>
));

export default Review;
