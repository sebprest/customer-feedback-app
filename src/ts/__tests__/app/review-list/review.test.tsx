import React from "react";
import renderer from "react-test-renderer";

import { FAKE_REVIEW } from "^/__tests__/__helpers__/fake-review";
import { Review } from "^/app/review-list/review";

describe("Review", () => {
  it("should render the details for the provided review ", () => {
    const reviewForm = renderer.create(<Review review={FAKE_REVIEW} />);

    expect(reviewForm).toMatchSnapshot();
  });
});
