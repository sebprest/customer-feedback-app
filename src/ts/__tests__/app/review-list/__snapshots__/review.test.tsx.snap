// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should render the details for the provided review  1`] = `
<div
  className="review"
>
  fake name
   (
  fake email
  ) - Rating: 
  5
  :
  <div
    className="review-comment"
  >
    fake comment
  </div>
</div>
`;
